src/Misc/DictionaryOperations.jl

```diff
@@ -0,0 +1,89 @@
+#=
+ MIT License
+
+ Copyright (c) 2010-present David A. Kopriva and other contributors: AUTHORS.md
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ --- End License
+=#
+
+#=
+    Some useful getters for a dictionary
+=#
+arrayRegex = r"(?<=\[).+?(?=\])"
+
+function realForKeyFromDictionary(key::AbstractString, d::Dict{String,Any})
+    v = d[key]
+    return parse(Float64,v)
+end
+
+function intForKeyFromDictionary(key::AbstractString, d::Dict{String,Any})
+    v = d[key]
+    return parse(Int64,v)
```

Comment:
Is this really an `Int64` or an `Int` (which is `Int32` on 32 bit systems and `Int64` on 64 bit sytems).

Reply:
For the record, you want to use the same definition on all systems, so it should be Int64. Otherwise, there can be different behavior on different systems. (I don't know how to make global aliases (as in C) or numeric KINDs (Fortran) in Julia to define project-wide numeric types.)

Reply:
That's what I was asking - so it should be `Int64`. Global aliases in Julia are just
```julia
const MySpecialRealType = Float64
```

Reply:
Unless it's an API thing or the need to store numbers greater than 2 billion, I recommend to stay with `Int` for user sanity. Anything else prompts the question "why this particular format" and is IMHO not necessary here.

Unrelated to this question: What is the purpose of these getter functions? Wouldn't it be easier (if it is possible) to just store the parsed values in the dict and then return directly the relevant data without conversion?

Reply:
And no one will ever need more than 640K of RAM. If I had known of a good way to define numeric types, as I do in C or fortran, I would have done that and then one would be guaranteed uniform behavior. I've lived through the transition from 16->32->64 bits,(used to be fun to give HW problems where depending on what machine students used they would get completely different answers) and it is a royal pain if you don't have parameterized numerics. There are standard ways to do this in other languages. The Julia docs never mentioned it, that I saw.

Reply:
On the unrelated question, it is done this way in HOHQMesh, too. It allows for the control file parser to be ignorant of the numeric data types being stored, and the actual data is accessed through lazy evaluation by entities that know what numeric type they want. It also bypasses conversion of string (from the data file) to numeric to string if it is not necessary. Alternatively, the data type would have to be encoded (or worse, use binary) and then the control file would not be human readable or editable. This dictionary approach makes it very easy to add new capability without re-writing code. When bottom topography, sweeping or extrusion were added to HOHQmesh, there was nothing that had to be done to the control file reader.